src/utils/Database/LSMTree/Generators.hs

```diff
@@ -506,3 +517,53 @@ chunkSizeUB = 20
 
 chunkSizeInvariant :: ChunkSize -> Bool
 chunkSizeInvariant (ChunkSize csize) = chunkSizeLB <= csize && csize <= chunkSizeUB
+
+{-------------------------------------------------------------------------------
+  Serialised keys/values/blobs
+-------------------------------------------------------------------------------}
+
+instance Arbitrary RawBytes where
+  arbitrary = genRawBytes >>= genSlice
+  shrink rb = shrinkRawBytes rb ++ shrinkSlice rb
+
+genRawBytesN :: Int -> Gen RawBytes
+genRawBytesN n = RawBytes . P.fromList <$> QC.vectorOf n arbitrary
+
+genRawBytes :: Gen RawBytes
+genRawBytes = RawBytes . P.fromList <$> QC.listOf arbitrary
+
+genRawBytesSized :: Int -> Gen RawBytes
+genRawBytesSized n = QC.resize n genRawBytes
+
+shrinkRawBytes :: RawBytes -> [RawBytes]
+shrinkRawBytes (RawBytes pvec) = [ RawBytes (P.fromList ws)
+                                 | ws <- QC.shrink (P.toList pvec) ]
+
+genSlice :: RawBytes -> Gen RawBytes
+genSlice (RawBytes pvec) = do
+    n <- QC.chooseInt (0, P.length pvec)
+    m <- QC.chooseInt (0, P.length pvec - n)
+    pure $ RawBytes (P.slice m n pvec)
+
+shrinkSlice :: RawBytes -> [RawBytes]
+shrinkSlice (RawBytes pvec) =
+    [ RawBytes (P.slice m n pvec)
+    | n <- QC.shrink (P.length pvec)
+    , m <- QC.shrink (P.length pvec - n)
+    ]
+
+deriving newtype instance Arbitrary SerialisedKey
+
+deriving newtype instance Arbitrary SerialisedValue
+
+instance Arbitrary SerialisedBlob where
+  arbitrary = SerialisedBlob <$> genRawBytes
+  shrink (SerialisedBlob rb) = SerialisedBlob <$> shrinkRawBytes rb
```

Comment:
I'll leave a comment. The reason I chose not to use `shrinkSlice` is because I don't think we're going to slice blobs from pages, but instead just read the number of bytes that we need directly